packages/contracts/contracts/provider/INodeProvider.sol

```diff
@@ -0,0 +1,40 @@
+// SPDX-License-Identifier: Apache-2.0
+/// @title Payment DApp Factory interface
+pragma solidity ^0.8.13;
+
+import {CartesiDApp} from "@cartesi/rollups/contracts/dapp/CartesiDApp.sol";
+import {ICartesiDApp} from "@cartesi/rollups/contracts/dapp/ICartesiDApp.sol";
+import {IConsensus} from "@cartesi/rollups/contracts/consensus/IConsensus.sol";
+
+import {IVault} from "../payment/IVault.sol";
+
+interface INodeProvider is IVault {
+    /// @notice Register a dapp with the node provider
+    /// @param dapp The dapp to be registered
+    /// @param location The cartesi machine snapshot location
+    function register(ICartesiDApp dapp, string memory location) external;
+
+    /// @notice Sets a name for the provider using ENS service
+    /// @param _name The name of the provider
+    function setName(string memory _name) external;
+}
+
+interface IReaderNodeProvider is INodeProvider {}
+
+interface IValidatorNodeProvider is IReaderNodeProvider {
+    /// @notice The consensus wired to the provider
+    function consensus() external view returns (IConsensus);
+
+    /// @notice Deploy a new application
+    /// @param _dappOwner The address that should own the DApp
+    /// @param _templateHash The hash of the initial state of the Cartesi Machine
+    /// @param _location The cartesi machine snapshot location
```

Comment:
```suggestion
    /// @param _location The Cartesi machine snapshot location
```

Reply:
I have a lot of `cartesi`. Are you sure I have to capitalize all occurrences of `cartesi` to `Cartesi`?
Is it `Cartesi Machine` or `Cartesi machine`?

Reply:
"Cartesi" is a proper noun.

Reply:
As for "machine", apparently we should capitalize it.